File: Primera parte/Método de Punto Fijo/Python/metodoPuntoFijo.py
def metodoPuntoFijo( x0, e1, e2, yi ):
   "Implementa el método de Punto Fijo"
   salida = [];
   x = testFuncion(x0, yi);
   salida.append(x);
   
   while (True):
   	if (testFuncion(x, yi) < (e2/100)):
   		break;
   	if (abs(x-x0) < (e1/100)):
   		break;
   	x0 = x;
   	x = testFuncion(x0,yi);
   	salida.append(x);
   return salida;

def testFuncion(x, yi):
   return (x**3 - yi) / 4;

print(metodoPuntoFijo(0.05, 0.005,0.005, 0));

File: Primera parte/Método de Punto Fijo/C/metodopuntofijo.c
#include <stdio.h>
#include <math.h>

int index = 0;

float testFuncion(float x, float yi) {
    return yi / (x*x) - 4;
}

float * metodoPuntoFijo(float x0, float e1, float e2, float yi) {
    
    static float salida[10000]; 
    float x = testFuncion(x0, yi);
	salida[index] = x;
    index++;
    while (1) {
		
		if (fabs(testFuncion(x, yi)) < (e2/100)) {
			break;
		}
		
		if (fabs(x - x0) < (e1/100)) {
			break;
		}
		
		x0 = x;
		x = testFuncion(x0, yi);
		salida[index] = x;
		index++;
		
    }    
    return salida;
}


int main () {
    float *salida = metodoPuntoFijo(100, 0.5,0.5, 2);
    int n = index;
    int i = 0;
    for (i = 0; i < n ; i++) {
        printf("%f\r\n", salida[i]);
    }
}